Feature/1-Maven-ize the build

nderwin opened the pull request:
This pull request uses Maven for producing the build artifacts.  It consists of a parent pom that sits above the main source code and provides common configuration for the MegaMek artifact.  This parent pom will be useful later for splitting the MegaMek.jar down into it's individual components for re-use (client, server, map editor, data library).

The pom in the megamek module will build the executable MegaMek.jar file, along with copying the needed data, docs, and mmconf directories and startup.sh script into the target folder.  The library dependencies are placed into target/lib, and are referenced as such in the MANIFEST file in the main jar.

Additionally, the assembly plugin is used to create a .tgz distribution of all of the necessary files for running the application.

nderwin committed:
added a project level Maven repo to house the TinyXML dependency
added a symbolic link to the data directory so the unit tests will run
updated the .gitignore file for the target directory and any NetBeans files
created the pom file that will compile and run the unit tests in place, then copy the data, docs, and mmconf directories to target, along with the startup.sh file so it will run in-place in target

nderwin committed:
Merge branch 'development' into maven-build

nderwin committed:
added the antrun plugin to put the executable permission back on startup.sh after the resources plugin removes it when copying the file to target
added the assembly plugin that takes the items from target and bundles them into a .tgz distribution file

nderwin committed:
Merge branch 'development' into maven-build

nderwin committed:
properly configured the antrun plugin
added an antrun execution to generate the equipment.txt file
added a manifest entry to make the MegaMek.jar file executable
silenced the warnings about long file names

nderwin committed:
Merge branch 'development' into maven-build

nderwin committed:
moved the pom, internal repo, and assembly files down a directory in preparation for making a parent pom in the top level directory
Task #1 - Maven-ize the build

nderwin committed:
moved the pom, internal repo, and assembly files down a directory in preparation for making a parent pom in the top level directory

nderwin committed:
added parent pom in preparation for more modularization
moved some of the config from the lower pom to the parent
moved the version string into a properties file (filtered by Maven)
Task #1 - Maven-ize the build

nderwin committed:
the version should be final, it's a constant
Task #1 - Maven-ize the build

djmbm commented:
While this is a great idea (Maven is awesome), it will not be accepted into the core MM project if you were to submit a pull request that way. I won't go into the long diatribe as to why, just letting you know ahead of time so we don't all waste time with such a pull request.

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
initial checkstyle config, with a minimal set of checks enabled
Task #3 - Add checkstyle to the build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'feature/1-Maven-ize-the-build' into feature/3-Add-checkstyle-to-the-build

nderwin committed:
removed some more style checks
fixed some build warnings

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'feature/1-Maven-ize-the-build' into feature/3-Add-checkstyle-to-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'feature/1-Maven-ize-the-build' into feature/3-Add-checkstyle-to-the-build

nderwin committed:
minimal rule set now established that will pass the current code from upstream
Task #3 - Add checkstyle to the build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'feature/1-Maven-ize-the-build' into feature/3-Add-checkstyle-to-the-build

nderwin committed:
Merge pull request #6 from nderwin/feature/3-Add-checkstyle-to-the-build

Feature/3 add checkstyle to the build

nderwin committed:
major overhaul to the build process based on the packaging_local.xml file
added 3 assembly descriptors for producing the Mac, *nix, and Windows distribution archives
moved some of the steps to better points in the build lifecycle
added some more compiler attributes to the parent pom

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

Conflicts:
	megamek/src/megamek/MegaMek.java

nderwin committed:
re-enable Checkstyle

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

Conflicts:
	megamek/src/megamek/MegaMek.java

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
updated the pom version from upstream

nderwin committed:
updated the pom version from the upstream version

nderwin committed:
fixed a typo in the local repo location

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
added testResources definition to pull test resources from the unittests directory for now

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
Merge branch 'development' into feature/1-Maven-ize-the-build

nderwin committed:
TinyXML has been removed from the upstream, so we don't need the project level repo or dependency anymore
Task #1 - Maven-ize the build
